doc/decisions/error_message.md

```diff
@@ -0,0 +1,137 @@
+#Error message & handling concept
+
+## Problem
+
+The current error concept has disadvantages in following regards:
+
+- Too verbose error message
+- A lot of redundant errors
+- Hard to manage specification file
+- No senseful way for application developers to use error codes from elektra
+
+This concept should improve useability for end users and developers.
+
+## Constraints
+
+- Error numbers must stay because they are more reliable to check against than strings
+
+## Assumptions
+
+## Considered Alternatives
+
+- Removing the specification file without requiring error numbers
```

Comment:
Here can you be much more concrete after you looked into other projects. Basically every project you look into will be a "considered alternative".